packages/amplify-e2e-tests/src/__tests__/api_2.test.ts

```diff
@@ -427,4 +427,23 @@ describe('amplify add api (REST)', () => {
       expect(unauthPolicies[i].PolicyName).toMatch(/PolicyAPIGWUnauth\d/);
     }
   });
+
+  it('migrates malformed project files during push', async () => {
+    await initJSProjectWithProfile(projRoot, {});
+    await addFunction(projRoot, { functionTemplate: 'Hello World' }, 'nodejs');
+    await addRestApi(projRoot, { existingLambda: true, restrictAccess: true });
+
+    const apisDirectory = path.join(projRoot, 'amplify', 'backend', 'api');
+    const apis = readdirSync(apisDirectory);
+    const apiName = apis[0];
+    const apiDirectory = path.join(apisDirectory, apiName);
+    const cfnTemplateFile = path.join(apiDirectory, `${apiName}-cloudformation-template.json`);
+    const cfnTemplate = JSON.parse(readFileSync(cfnTemplateFile, 'utf8'));
+
+    // The ApiId output is required, and will be added automatically if it is missing.
+    cfnTemplate.Outputs.ApiId = undefined;
+
+    writeFileSync(cfnTemplateFile, JSON.stringify(cfnTemplate));
+    await amplifyPushUpdate(projRoot);
```

Comment:
Does this only happen for the output? or do we generate the full Cfn again?

Reply:
The full CFN is not regenerated, just the `ApiId` output is added.